### PR TITLE
chore(mergequeue): add configuration

### PR DESCRIPTION
**Pull request by wdhif**
### What does this PR do?

This PR adds the same Merge Queue configuration as the [Helm Chart](https://github.com/DataDog/helm-charts) repository.

### Motivation

This is needed to be able to merge using Merge Queue.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

**Review by clamoriniere — COMMENTED** [review]
change looks good but could you please split the PR in 2, because mixing tooling config update with code change event it is documentation.

**Comment by wdhif** [comment]
> change looks good but could you please split the PR in 2, because mixing tooling config update with code change event it is documentation.

Sure! Documentation commit has been moved here: https://github.com/DataDog/datadog-operator/pull/1194

**Review by clamoriniere — APPROVED** [review]

File: repository.datadog.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  - container-helm-chart-maintainers
+  - container-integrations
+  - container-t2
+  - Synthetics
+  - Documentation
+  - Observability Pipelines
+  - Telemetry and Analytics
+  - Vector
+github_users_restrictions:
+  - cahillsf
```
Comment (clamoriniere) [inline repository.datadog.yml:18]:
```suggestion
```

File: repository.datadog.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - agent-all
+  - container-app
+  - container-ecosystems
+  - container-helm-chart-maintainers
+  - container-integrations
+  - container-t2
+  - Synthetics
+  - Documentation
+  - Observability Pipelines
+  - Telemetry and Analytics
```
Comment (clamoriniere) [inline repository.datadog.yml:15]:
```suggestion
  - container-helm-chart-maintainers
  - container-integrations
  - container-t2
  - Documentation
```

**Pull request merged**